app/controllers/concerns/formatting_concern.rb

```diff
@@ -36,6 +36,10 @@ def pass_records_through_xml_parser(records)
     end
   end
 
+  def valid_barcode(barcode='')
+    barcode =~ /^(32101[0-9]{9}|PULTST[0-9]{5})$/
```

Comment:
Could the method be renamed `valid_barcode?`.
Also the method should either return true or false, not 0 or nil. Try this to get a boolean response:
```
(barcode =~ /^(32101[0-9]{9}|PULTST[0-9]{5})$/) == 0
```